napari/layers/_labels_layer/model.py

```diff
@@ -521,58 +461,43 @@ def _interp_coords(self, old_coord, new_coord):
 
         return coords
 
-    def get_label(self, position, indices):
+    def get_value(self):
         """Returns coordinates, values, and a string for a given mouse position
         and set of indices.
 
-        Parameters
-        ----------
-        position : sequence of two int
-            Position of mouse cursor in canvas.
-        indices : sequence of int or slice
-            Indices that make up the slice.
-
         Returns
         ----------
-        coord : sequence of int
-            Position of mouse cursor in image coordinates.
-        label : int
-            Value of the label image at the coord.
+        coord : sequence of float
+            Position of mouse cursor in data.
+        value : int or float or sequence of int or float
+            Value of the data at the coord.
         """
-        transform = self._node.canvas.scene.node_transform(self._node)
-        pos = transform.map(position)
-        pos = [np.clip(pos[1], 0, self._image_view.shape[0]-1),
-               np.clip(pos[0], 0, self._image_view.shape[1]-1)]
-        coord = list(indices)
-        coord[-2] = pos[0]
-        coord[-1] = pos[1]
-        int_coord = copy(coord)
-        int_coord[-2] = int(round(coord[-2]))
-        int_coord[-1] = int(round(coord[-1]))
-        label = self._image_view[tuple(int_coord[-2:])]
-
-        return coord[-self.image.ndim:], label
-
-    def get_message(self, coord, label):
+        coord = np.round(self.coordinates).astype(int)
+        coord[-2] = np.clip(coord[-2], 0, self._image_view.shape[0]-1)
+        coord[-1] = np.clip(coord[-1], 0, self._image_view.shape[1]-1)
+
+        value = self._image_view[int(round(coord[-2])), int(round(coord[-1]))]
```

Comment:
Why is more rounding and casting needed here?

Reply:
fixed